### PR TITLE
Enhanced migrate views task to support views created with explicit column list

### PR DESCRIPTION
**Pull request by FastLee**
Added a lookup based on "SHOW CREATE TABLE" to retrieve the list of columns from the create script.

### CODE REVIEW

File: src/databricks/labs/ucx/hive_metastore/table_migrate.py [hunk 1]
```diff
@@ -202,7 +203,14 @@ def _get_view_update_sql(self, src_table: Table, rule: Rule) -> str:
         new_view_text = ViewToMigrate.get_view_updated_text(
             src_table.view_text, self._migration_status_refresher.index(), src_table.database
         )
-        return f"CREATE VIEW IF NOT EXISTS {escape_sql_identifier(rule.as_uc_table_key)} AS {new_view_text};"
+        # Getting columns definition from the show create table output
+        create_statement = self._backend.fetch(f"SHOW CREATE TABLE {src_table.safe_sql_key}")
+        create_str = next(iter(create_statement))["createtab_stmt"]
```
Comment (nfx) [inline src/databricks/labs/ucx/hive_metastore/table_migrate.py:208]: Can't we feed it to get_view_updated_text to get rid of that potentially buggy "get view columns?

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/hive_metastore/table_migrate.py (hunk 1 above)
Comment (FastLee) [reply src/databricks/labs/ucx/hive_metastore/table_migrate.py:208]: I looked into it.
It is a 50/50 split.
It has the tblproperties that we will have to strip out. And we will have to translate the name of the view (not a big challenge).

Comment (nfx) [reply src/databricks/labs/ucx/hive_metastore/table_migrate.py:208]: Shouldn't we keep the properties while migrating? What properties are there?

Changing view name is just a swap of identifier, which is easy

Comment (FastLee) [reply src/databricks/labs/ucx/hive_metastore/table_migrate.py:208]: We shouldn't.
Most properties are irrelevant and auto generated.

**Pull request merged**